rfcs/convergence/file-naming.md

```diff
@@ -0,0 +1,152 @@
+| index.tsx           | index.ts            |                                                                                     | No changes from current                                                                                                                                                                                                                                                                                                   |
+| Example.tsx         | example.ts          | `export const Example = () => {..}`                                                 | To enforce rendering logic only within `.view.` type file/s. TS will enforce this by default. Not extremely picky on this one, we can use `.tsx` as well to match extension with test/story - priority is to be consistent                                                                                                |
+| Example.test.tsx    | example.spec.tsx    | `describe->it`                                                                      | Integration test on JSDOM level (jest) for whole component                                                                                                                                                                                                                                                                |
+| storybook (NONE)    | example.stories.tsx | `export const Example = () => {..}`                                                 | Specific file type for Storybook consumption. Main "starting" point for inner loop - daily development.                                                                                                                                                                                                                   |
+| mdx (NONE)          | example.stories.mdx | ---                                                                                 | Markdown(MDX) for consumer facing documentation(mainly for storybook at the moment, but can be transformed to any other tool that can parse MDX).                                                                                                                                                                         |
+| renderExample.tsx   | example.view.tsx    | `export const render = () => {..}`                                                  | `.render` might work as well, to match symbol name.                                                                                                                                                                                                                                                                       |
```

Comment:
A function should not be named as `render`, it should be `renderComponentName` as we should avoid collisions in consumer imports. For example:

```jsx
import { render } from 'react-dom'
import { render } from '@fluentui/react-button'
import { render } from '@fluentui/react-menu'
```

It also breaks DX on our side as if search will be used how many matches for `render` you will get?

Reply:
For me, it's confusing to name files as `example.view.tsx` when a single export is `renderExample` function. It's not a common thing in React community:
- https://github.com/mui-org/material-ui/tree/next/packages/material-ui-utils/src
- https://github.com/adobe/react-spectrum/tree/main/packages/%40react-spectrum/utils/src

Reply:
> `renderComponentName`

That's a typo, thx.

> A function should not be named as `render`, it should be `renderComponentName` as we should avoid collisions in consumer imports. For example:

FWIW that's not true :) consumer can name imported symbols anyhow he wants. 

> It also breaks DX on our side as if search will be used how many matches for render you will get?

maybe out of topic but my understanding was that those renderers are meant to be used solely per package/component

Reply:
"consumer can name imported symbols anyhow he wants." - from my experience with navigating relatively large code bases - the more specificity you have in names and the more consistency there is between file and export names, the better - because then you can easily combine fulltext search and file name search

In 99% of the cases probably yes, renderers will be used only in the package and component. But if the consumer wants to extend the component and add/replace hooks in it, he can still import and reuse the `render` function.